### PR TITLE
Adding source and destination info to context.

### PR DESCRIPTION
**Pull request by achettyiitr**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: warehouse/index.js [hunk 1]
```diff
@@ -463,6 +463,19 @@ function storeRudderEvent(utils, message, output, columnTypes, options) {
   ]
 */
 
+/*
+* Adds source and destination specific information into context
+* */
+function enhanceContextWithSourceDestInfo(context, metadata) {
+  if (!context || !metadata) {
```
Comment (psrikanth88) [inline warehouse/index.js:470]: set empty context object {} if not present

**Review by psrikanth88 — APPROVED** [review]

File: warehouse/index.js [hunk 2]
```diff
@@ -475,6 +488,9 @@ function processWarehouseMessage(message, options) {
     message.messageId = `auto-${randomID}`;
   }
 
+  // Adding source and destination specific information.
+  enhanceContextWithSourceDestInfo(message.context, options.metadata)
```
Comment (psrikanth88) [inline warehouse/index.js:492]: add a flag around this

File: warehouse/index.js (hunk 1 above)
Comment (achettyiitr) [reply warehouse/index.js]: Done.

File: warehouse/index.js (hunk 2 above)
Comment (achettyiitr) [reply warehouse/index.js]: Done.

**Pull request merged**